lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -441,34 +451,31 @@ def set_3d_properties(self, zs, zdir):
         # just in case it is a scalarmappable with a colormap.
         self.update_scalarmappable()
         offsets = self.get_offsets()
-        if len(offsets) > 0:
-            xs, ys = list(zip(*offsets))
-        else:
-            xs = []
-            ys = []
-        self._offsets3d = juggle_axes(xs, ys, np.atleast_1d(zs), zdir)
+        offsets = np.hstack([offsets,
+                             (np.ones(len(offsets)) * zs)[:, np.newaxis]])
+        self._offsets3d = juggle_axes_vec(offsets, zdir).T
         self._facecolor3d = self.get_facecolor()
         self._edgecolor3d = self.get_edgecolor()
         self.stale = True
 
     def do_3d_projection(self, renderer):
         xs, ys, zs = self._offsets3d
-        vxs, vys, vzs, vis = proj3d.proj_transform_clip(xs, ys, zs, renderer.M)
+        vxyzis = proj3d.proj_transform_clip(xs, ys, zs, renderer.M)
 
-        fcs = (zalpha(self._facecolor3d, vzs) if self._depthshade else
+        fcs = (zalpha(self._facecolor3d, vxyzis[2]) if self._depthshade else
                self._facecolor3d)
         fcs = mcolors.to_rgba_array(fcs, self._alpha)
         self.set_facecolors(fcs)
 
-        ecs = (zalpha(self._edgecolor3d, vzs) if self._depthshade else
+        ecs = (zalpha(self._edgecolor3d, vxyzis[2]) if self._depthshade else
                self._edgecolor3d)
         ecs = mcolors.to_rgba_array(ecs, self._alpha)
         self.set_edgecolors(ecs)
-        PathCollection.set_offsets(self, list(zip(vxs, vys)))
+        PathCollection.set_offsets(self, vxyzis[0:2].T)
 
-        if vzs.size > 0 :
-            return min(vzs)
-        else :
+        if len(vxyzis) > 0:
```

Comment:
The condition should be `len(vxyzis[2]) > 0`. You'd better to make `vzs = vxyzis[2]` (or `_, _, vzs, _ = vxyzis = proj3d....`) and then use `vzs` here and above.